helm/charts/hydra/templates/rbac.yaml

```diff
@@ -0,0 +1,32 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "hydra.fullname" . }}
+  namespace:  {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "hydra.fullname" . }}
+  namespace:  {{ .Release.Namespace }}
+rules:
+  - apiGroups: ["*"]
+    resources: ["pods", "deployments", "services", "configmaps", "serviceaccounts"]
```

Comment:
Why do we need access to pods, deployments, services, configmaps and serviceaccounts? And why list, get, watch, create, update, patch, delete? Wouldn't it be much better to just have a very minimalistic service account? It seems like this permission list is very exhaustive and not neccessary for hydra to work

Reply:
Hi @aeneasr 

Yes, but with Role the serviceaccount will have effect at a namespace level which is less than a ClusterRole. However I agree we should reduce the number of verbs.

For this particular use-case where we have PodSecurityPolicy I think this should be the possible rules:
```
- apiGroups: ["*"]
  resources: ["pods", "deployments"]
  verbs: ["get", "list", "watch", "create", "update"]
- apiGroups: ["*"]
  resources: [serviceaccounts]
  verbs: ["get", "list", "watch"]
```

Reply:
Hi @aeneasr 

Sure you followed the conversation in the public slack channel. Here is a particular scenario involving istio  and where we have to define a PodSecurityPolicy. Actually we define PodSecurityPolicy in all our kuberntes applications deployed in our cluster not only hydra.

Reply:
I believe it would make more sense to have the service account name configurable, that way you can "mount" any service account you'd like. As said before, the "raw" pods of Hydra do not need any of this, so - imo - this shouldn't be here.